src/mongoose_hooks.erl

```diff
@@ -1036,23 +1036,15 @@ mam_lookup_messages(HookServer, InitialValue, Params) ->
                             [HookServer, Params]).
 
 %%% @doc The `mam_archive_message' hook is called in order to store the message in the archive.
--spec mam_archive_message(HookServer, InitialValue, MessageID, ArchiveID, OwnerJID, RemoteJID, SenderJID, OriginID, Dir, Packet) ->
+-spec mam_archive_message(HookServer, InitialValue, Params) ->
     Result when
     HookServer :: jid:lserver(),
     InitialValue :: ok,
-    MessageID :: mod_mam:message_id(),
-    ArchiveID :: undefined | mod_mam:archive_id(),
-    OwnerJID :: jid:jid(),
-    RemoteJID :: jid:jid(),
-    SenderJID :: jid:jid(),
-    OriginID :: binary() | none,
-    Dir :: incoming | outgoing,
-    Packet :: term(),
+    Params :: mod_mam:archive_message_params(),
     Result :: ok | {error, timeout}.
-mam_archive_message(HookServer, InitialValue, MessageID, ArchiveID, OwnerJID, RemoteJID, SenderJID, OriginID, Dir, Packet) ->
+mam_archive_message(HookServer, InitialValue, Params) ->
     ejabberd_hooks:run_fold(mam_archive_message, HookServer, InitialValue,
```

Comment:
There might be custom MAM backend in the wild relying on the old `mam_archive_message` hook. Such custom module will not work without changes after upgrading to newest MongooseIM. Because of that, I think it'd be good to put information about the hook change in new migration guide `doc/migrations/3.7.0_3.7.0++.md` (as usual, the file name will be changed as part of the release process).